### PR TITLE
docker_swarm_service: Documentation fixes

### PR DESCRIPTION
**Pull request by hannseman**
##### SUMMARY

Fixes a number of issues:
* `mode` choices is now validated and documented. Documentation about recreation of service is now correctly under this option and not `publish.mode`
* Update and extend examples
* Some test fixes
* Update module return value sample with correct fields
* Add `secrets` and `users` to module return value

BTW @felixfontein I love these new fast tests!

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

docker_swarm_service

### CODE REVIEW

**Comment by felixfontein** [comment]
This PR is pretty hard to read. Can you split it up into two PRs:
 1. the first one with everything but docs reordering,
 2. the second one only docs reordering (but no changes)?
That would make reviewing the PRs much easier.

File: lib/ansible/modules/cloud/docker/docker_swarm_service.py [hunk 1]
```diff
@@ -461,7 +465,6 @@
       mode:
         description:
           - What publish mode to use.
-          - Service will be removed and recreated when changed.
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/docker/docker_swarm_service.py:464]: Is this no longer true for `publish` mode?

**Comment by hannseman** [comment]
@felixfontein good idea. I'll create a PR with the reordering once this is merged.

File: lib/ansible/modules/cloud/docker/docker_swarm_service.py (hunk 1 above)
Comment (hannseman) [reply lib/ansible/modules/cloud/docker/docker_swarm_service.py:464]: This was never true. In https://github.com/ansible/ansible/pull/52634/ I added that string to the wrong `mode`-option.

Comment (hannseman) [reply lib/ansible/modules/cloud/docker/docker_swarm_service.py:464]: It's supposed to be documented on the actual `mode`-option and not `publish.mode`.

**Review by felixfontein — APPROVED** [review]
shipit

**Review by felixfontein — APPROVED** [review]
shipit

**Comment by hannseman** [comment]
@felixfontein thank you for reviewing. @sivel thank you for merging.

**Comment by felixfontein** [comment]
@hannseman and thanks for the PR! :) Looks like #53482 needs a rebase now...

**Pull request merged**